web_external/Datasets/UploadDatasetMetadataView.js

```diff
@@ -0,0 +1,248 @@
+// View for a collection of datasets in a select tag
+isic.views.UploadDatasetMetadataSelectDatasetView = isic.View.extend({
+    events: {
+        'change': 'datasetChanged'
+    },
+
+    initialize: function (options) {
+        this.listenTo(this.collection, 'reset', this.render);
+
+        this.render();
+    },
+
+    datasetChanged: function () {
+        this.trigger('changed', this.$('select').val());
+    },
+
+    render: function () {
+        // Destroy previous select2
+        var select = this.$('#isic-upload-dataset-metadata-dataset-select');
+        select.select2('destroy');
+
+        this.$el.html(isic.templates.uploadDatasetMetadataSelectDatasetPage({
+            models: this.collection.models
+        }));
+
+        // Set up select box
+        var placeholder = 'Select a dataset...';
+        if (!this.collection.isEmpty()) {
+            placeholder += ' (' + this.collection.length + ' available)';
+        }
+        select = this.$('#isic-upload-dataset-metadata-dataset-select');
+        select.select2({
+            placeholder: placeholder
+        });
+        select.focus();
+
+        return this;
+    }
+});
+
+isic.views.UploadDatasetMetadataView = isic.View.extend({
```

Comment:
This `UploadDatasetMetadataView` is the only thing in this PR that I haven't reviewed yet, just because it could change substantially if we get rid of the `UploadDatasetMetadataSelectDatasetView`.